### ISSUE TITLE
Undefined symbol: curl_multi_ce on php8.0

### ISSUE THREAD

**Issue by metaxy** [DataDog/dd-trace-php#1157]
### Bug description
The extension ddtrace won't load because of `undefined symbol: curl_multi_ce`

```
Warning: PHP Startup: Unable to load dynamic library 'ddtrace.so' (tried: /opt/dd-trace-php-master/modules/ddtrace.so (/opt/dd-trace-php-master/modules/ddtrace.so: undefined symbol: curl_multi_ce), /opt/dd-trace-php-master/modules/ddtrace.so.so (/opt/dd-trace-php-master/modules/ddtrace.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0
```

### PHP version
```
PHP 8.0.2 (cli) (built: Feb  5 2021 04:58:33) ( NTS )
Copyright (c) The PHP Group
Zend Engine v4.0.2, Copyright (c) Zend Technologies
    with Zend OPcache v8.0.2, Copyright (c), by Zend Technologies
```

### Installed extensions
```
[PHP Modules]
bcmath
bz2
calendar
Core
ctype
curl
date
dba
dom
enchant
exif
fileinfo
filter
ftp
gd
gettext
gmp
hash
iconv
imap
intl
json
ldap
libxml
mbstring
memcached
mysqli
mysqlnd
odbc
openssl
pcntl
pcre
PDO
pdo_mysql
PDO_ODBC
pdo_pgsql
pdo_sqlite
pgsql
Phar
posix
pspell
readline
redis
Reflection
session
SimpleXML
soap
sockets
sodium
SPL
sqlite3
standard
sysvmsg
sysvsem
sysvshm
tidy
tokenizer
xml
xmlreader
xmlwriter
xsl
Zend OPcache
zip
zlib

[Zend Modules]
Zend OPcache
```

### OS info
```
NAME="Ubuntu"
VERSION="18.04.5 LTS (Bionic Beaver)"
PRETTY_NAME="Ubuntu 18.04.5 LTS"
VERSION_ID="18.04"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```

#### Output of phpinfo() (ddtrace >= 0.47.0)

None. Because it is not loaded.

But maybe it significant how its compiled:
![Screenshot_20210225_101547](https://user-images.githubusercontent.com/82704/109130852-7a6e3a80-7752-11eb-91ab-6673c1fcc324.jpeg)

### Upgrading info

I am using php 8.0 on plesk (compiled by them). First i used version 0.55 and got the same error. But then i saw the pull request #1144. So i used master to use this bug fix. But it didn't help.

```bash
sudo apt install libcurl4-openssl-dev
cd /opt
wget https://github.com/DataDog/dd-trace-php/archive/master.zip
unzip master.zip
cd dd-trace-php-master 
export DD_TRACE_PHP_BIN=/opt/plesk/php/8.0/bin/php
/opt/plesk/php/8.0/bin/phpize
./configure --enable-ddtrace --with-php-config=/opt/plesk/php/8.0/bin/php-config
make
make install
```

With php 7.4 it was working fine.

**Comment by SammyK** [comment]
Thank you for the report @metaxy. We are investigating this issue and will update this thread when we have more information to report.

**Comment by SammyK** [comment]
@metaxy We have been able to replicate this issue and are working on a fix in #1181. This should be released in the next version of the tracer (`0.57.0`). Thanks again for the report!

**Comment by metaxy** [comment]
Thank you for your work.

Since `0.57.0` is released and this issue is mentioned in the release log: The build does work but compiling from source works fine:

# Installing from build
- removed 98-ddtrace.ini and 99-ddtrace-custom.ini
- removed /opt/datadog-php
```bash
> wget https://github.com/DataDog/dd-trace-php/releases/download/0.57.0/datadog-php-tracer_0.57.0_amd64.deb -O datadog-php-tracer.deb
2021-04-19 10:10:07 (17.1 MB/s) - ‘datadog-php-tracer.deb’ saved [23112630/23112630]
> sha1sum datadog-php-tracer.deb
1dc3790cbf49833b56b696f3a9370e4200356f3e  datadog-php-tracer.deb
```

```bash
> dpkg -i datadog-php-tracer.deb
(Reading database ... 182965 files and directories currently installed.)
Preparing to unpack datadog-php-tracer.deb ...
Unpacking datadog-php-tracer (0.57.0) over (0.57.0) ...
Setting up datadog-php-tracer (0.57.0) ...

This server is powered by Plesk.

Run the 'plesk login' command and log in by browsing either of the links received in the output.
Use the 'plesk' command to manage the server. Run 'plesk help' for more info.

### PHP version
PHP 8.0.3 (cli) (built: Mar  5 2021 08:15:38) ( NTS )
Copyright (c) The PHP Group
Zend Engine v4.0.3, Copyright (c) Zend Technologies
    with Zend OPcache v8.0.3, Copyright (c), by Zend Technologies
### Installing Datadog PHP tracing extension (ddtrace)
###
### Logging /opt/plesk/php/8.0/bin/php -i to a file
###
### Creating ddtrace.ini
### 

; ***** DO NOT EDIT THIS FILE *****
; To overwrite the INI settings for this extension, edit
; the INI file in this directory called "ddtrace-custom.ini"

[datadog]
extension=/opt/datadog-php/extensions/ddtrace-20200930.so
ddtrace.request_init_hook=/opt/datadog-php/dd-trace-sources/bridge/dd_wrap_autoloader.php
### ddtrace.ini created
###
### Created empty /opt/datadog-php/etc/ddtrace-custom.ini
###
### Linking ddtrace.ini for supported SAPI's
### 

### Found SAPI config directory: /opt/plesk/php/8.0/etc/php.d
### Linking ddtrace.ini to /opt/plesk/php/8.0/etc/php.d/98-ddtrace.ini
### Linking ddtrace-custom.ini to /opt/plesk/php/8.0/etc/php.d/99-ddtrace-custom.ini
###
PHP Warning:  PHP Startup: Unable to load dynamic library '/opt/datadog-php/extensions/ddtrace-20200930.so' (tried: /opt/datadog-php/extensions/ddtrace-20200930.so (/opt/datadog-php/extensions/ddtrace-20200930.so: undefined symbol: curl_multi_ce), /opt/plesk/php/8.0/lib/php/modules//opt/datadog-php/extensions/ddtrace-20200930.so.so (/opt/plesk/php/8.0/lib/php/modules//opt/datadog-php/extensions/ddtrace-20200930.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0
### Failed enabling ddtrace extension
###
### The extension has been installed but couldn't be enabled
### Try adding the extension manually to your PHP - php.ini - configuration file
### e.g. by adding following line: 
###
###     extension=/opt/datadog-php/extensions/ddtrace-20200930.so
###
### Note that your PHP API version must match the extension's API version
### PHP API version can be found using following command
###
###     /opt/plesk/php/8.0/bin/php -i | grep 'PHP API'
###


> /opt/plesk/php/8.0/bin/php --ri=ddtrace
PHP Warning:  PHP Startup: Unable to load dynamic library '/opt/datadog-php/extensions/ddtrace-20200930.so' (tried: /opt/datadog-php/extensions/ddtrace-20200930.so (/opt/datadog-php/extensions/ddtrace-20200930.so: undefined symbol: curl_multi_ce), /opt/plesk/php/8.0/lib/php/modules//opt/datadog-php/extensions/ddtrace-20200930.so.so (/opt/plesk/php/8.0/lib/php/modules//opt/datadog-php/extensions/ddtrace-20200930.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0
Extension 'ddtrace' not present.
```

# Build from source

```bash
sudo apt install libcurl4-openssl-dev
cd /opt
wget https://github.com/DataDog/dd-trace-php/archive/refs/tags/0.57.0.zip
unzip 0.57.0.zip
cd dd-trace-php-0.57.0 
/opt/plesk/php/8.0/bin/phpize
./configure --enable-ddtrace --with-php-config=/opt/plesk/php/8.0/bin/php-config
make
make install
```

After adding the ini file:

```bash
# /opt/plesk/php/8.0/bin/php --ri=ddtrace

ddtrace

Datadog PHP tracer extension
For help, check out the documentation at https://docs.datadoghq.com/tracing/languages/php/
(c) Datadog 2020

Datadog tracing support => enabled
Version => 0.57.0
DATADOG TRACER CONFIGURATION => {"date":"2021-04-19T08:20:43Z","os_name":"Linux tertius.ebtc-online.org 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64","os_version":"4.15.0-74-generic","version":"0.57.0","lang":"php","lang_version":"8.0.3","env":null,"enabled":true,"service":null,"enabled_cli":false,"agent_url":"http://localhost:8126","debug":false,"analytics_enabled":false,"sample_rate":1.000000,"sampling_rules":null,"tags":null,"service_mapping":null,"distributed_tracing_enabled":true,"priority_sampling_enabled":true,"dd_version":null,"architecture":"x86_64","sapi":"cli","ddtrace.request_init_hook":"/opt/dd-trace-php-0.57.0/bridge/dd_wrap_autoloader.php","open_basedir_configured":false,"uri_fragment_regex":null,"uri_mapping_incoming":null,"uri_mapping_outgoing":null,"auto_flush_enabled":false,"generate_root_span":true,"http_client_split_by_domain":false,"measure_compile_time":true,"report_hostname_on_root_span":false,"traced_internal_functions":null,"auto_prepend_file_configured":false,"integrations_disabled":null,"enabled_from_env":true,"opcache.file_cache":null}

                               Diagnostics                               
Diagnostic checks => passed

Directive => Local Value => Master Value
ddtrace.disable => Off => Off
ddtrace.request_init_hook => /opt/dd-trace-php-0.57.0/bridge/dd_wrap_autoloader.php => /opt/dd-trace-php-0.57.0/bridge/dd_wrap_autoloader.php
ddtrace.cgroup_file => /proc/self/cgroup => /proc/self/cgroup
```

**Comment by SammyK** [comment]
Thanks for the followup @metaxy. Our packages are built on a machine that has the curl extension enabled, so that might explain why this error still occurs when installing from one of the prebuilt packages. To confirm, would you mind verifying [that this patched `.deb`](https://391320-119990860-gh.circle-artifacts.com/0/datadog-php-tracer_1.0.0-nightly_amd64.deb) no longer gives the `undefined symbol: curl_multi_ce` error message?

**Comment by metaxy** [comment]
Yes, this worked fine.

**Comment by SammyK** [comment]
Thank you for confirming @metaxy. We have started working on a fix for this in #1209.

**Comment by SammyK** [comment]
This issue was fixed and released in `0.59.0`. Please let us know if you are still having issues with this after upgrading. Thank you again for the report @metaxy.
